pkg/products/monitoringspec/reconciler.go

```diff
@@ -225,6 +299,18 @@ func (r *Reconciler) reconcileMonitoring(ctx context.Context, serverClient k8scl
 			}
 		}
 	}
+	//TODO remove from here vvvvvvvvvvvvvvvvvvvvv
+	for _, object := range obsoleteObjects {
+		err := r.removeOldResources(ctx, serverClient, object.objectType, object.name, object.namespace)
+		if k8serr.IsNotFound(err) {
+			//ignore not found, We expect to hit this on every reconcile apart from the first after upgrade
```

Comment:
***Note to self:*** come up with some better way of confirming that the resources have been deleted